vowpalwabbit/gd.cc

```diff
@@ -1284,9 +1284,11 @@ base_learner* setup(VW::setup_base_i& stack_builder)
   if (g->adax && !all.weights.adaptive) THROW("Cannot use adax without adaptive");
 
   if (pow(static_cast<double>(all.eta_decay_rate), static_cast<double>(all.numpasses)) < 0.0001)
-    *(all.trace_message) << "Warning: the learning rate for the last pass is multiplied by: "
-                         << pow(static_cast<double>(all.eta_decay_rate), static_cast<double>(all.numpasses))
-                         << " adjust --decay_learning_rate larger to avoid this." << std::endl;
+  {
```

Comment:
some trace messages are replaced with logged messages (like here) but some are not (e.g. cs_active) what's the logic behind replacing some and not others? is it that if it's an error or warning it now goes to logger but otherwise stays on driver which can be set independently?

Reply:
I replaced trace messages that were obviously a warning or error, if I missed any please point them out. I left things that sort of just look like driver sort of updates. I'm not sure whether to move those to an info level log or not. Its still not exactly clear what exactly constitutes the driver output vs info